doc/introduction.md

```diff
@@ -1,16 +1,86 @@
 # Introduction
 
-RxJS is a library for composing asynchronous and event-based programs by using observable sequences. It provides one core type, the [Observable](./overview.html#observable), satellite types (Observer, Schedulers, Subjects) and operators inspired by [Array#extras](https://developer.mozilla.org/en-US/docs/Web/JavaScript/New_in_JavaScript/1.6) (map, filter, reduce, every, etc) to allow handling asynchronous events as collections.
-
 <span class="informal">Think of RxJS as lodash for events.</span>
 
-ReactiveX combines the [Observer pattern](https://en.wikipedia.org/wiki/Observer_pattern) with the [Iterator pattern](https://en.wikipedia.org/wiki/Iterator_pattern) and [functional programming with collections](http://martinfowler.com/articles/collection-pipeline/#NestedOperatorExpressions) to fill the need for an ideal way of managing sequences of events.
+```js
+// Normally you register events with
+const button = document.querySelector('button');
+button.addEventListener('click', () => console.log('Clicked!'));
+
+// Using RxJS you create an observable
+const button = document.querySelector('button');
+Rx.Observable.fromEvent(button, 'click')
+  .subscribe(() => console.log('Clicked!'));
+```
+
+## Keep it pure
+What makes RxJS powerful is its ability to produce values using pure functions. That means your code is less prone for errors.
+
+```js
+// Normally you would create an impure function, where other
+// pieces of your code can mess up your state
+let count = 0;
+const button = document.querySelector('button');
+button.addEventListener('click', () => console.log(`Clicked ${++count} times`));
+
+// Using RxJS you isolate the state
+const button = document.querySelector('button');
+Rx.Observable.fromEvent(button, 'click')
+  .scan(count => count + 1, 0)
+  .subscribe(count => console.log('Clicked ${count} times'));
+```
+
+The **scan** operator works just like **reduce**. It takes a value which is exposed to a callback. The returned value of the callback will then become the next value exposed the next time the callback runs.
+
+## Control the flow
+RxJS has a whole range of operators that helps you control how the events flow through your observables.
+
+```js
+// Only allow clicks every second
```

Comment:
Maybe add `// With Plain JS:` after this line, to show you're comparing what's below to the `// With RxJS:` part after that.